src/lib/metric-events.ts

```diff
@@ -9,6 +11,51 @@ const PROXY_REPOSITORY_CREATED = 'proxy_repository_created';
 const PROXY_FEATURES_FOR_TOKEN_TIME = 'proxy_features_for_token_time';
 const STAGE_ENTERED = 'stage-entered' as const;
 const EXCEEDS_LIMIT = 'exceeds-limit' as const;
+const REQUEST_ORIGIN = 'request_origin' as const;
+
+type MetricEvent =
+    | typeof REQUEST_TIME
+    | typeof DB_TIME
+    | typeof FUNCTION_TIME
+    | typeof SCHEDULER_JOB_TIME
+    | typeof FEATURES_CREATED_BY_PROCESSED
+    | typeof EVENTS_CREATED_BY_PROCESSED
+    | typeof FRONTEND_API_REPOSITORY_CREATED
+    | typeof PROXY_REPOSITORY_CREATED
+    | typeof PROXY_FEATURES_FOR_TOKEN_TIME
+    | typeof STAGE_ENTERED
+    | typeof EXCEEDS_LIMIT
+    | typeof REQUEST_ORIGIN;
```

Comment:
I think it could be nice to simplify this in the future. E.g. for feature flags in `experimental.ts` we have the literals in a single type instead of declaring a constant for each flag name.